tests/model_registry/rest_api/conftest.py

```diff
@@ -63,3 +89,128 @@ def updated_model_registry_resource(
+        yield mr
+
+
+@pytest.fixture(scope="class")
+def mysql_template_with_ca(model_registry_db_secret: Secret) -> dict[str, Any]:
+    """
+    Patches the MySQL template with the CA file path and volume mount.
+    Args:
+        model_registry_db_secret: The secret for the model registry's MySQL database
+    Returns:
+        dict[str, Any]: The patched MySQL template
+    """
+    mysql_template = get_model_registry_deployment_template_dict(
+        secret_name=model_registry_db_secret.name,
+        resource_name=DB_RESOURCES_NAME,
+    )
+    mysql_template["spec"]["containers"][0]["args"].append(f"--ssl-ca={CA_FILE_PATH}")
+    mysql_template["spec"]["containers"][0]["volumeMounts"].append({
+        "mountPath": CA_MOUNT_PATH,
+        "name": CA_CONFIGMAP_NAME,
+        "readOnly": True,
+    })
+    mysql_template["spec"]["volumes"].append({"name": CA_CONFIGMAP_NAME, "configMap": {"name": CA_CONFIGMAP_NAME}})
+    return mysql_template
+
+
+@pytest.fixture(scope="class")
+def deploy_secure_mysql_and_mr(
+    model_registry_namespace: str,
+    model_registry_db_secret: Secret,
+    model_registry_db_deployment: Deployment,
+    model_registry_mysql_config: dict[str, Any],
+    mysql_template_with_ca: dict[str, Any],
+) -> Generator[ModelRegistry, None, None]:
+    """
+    Deploys MySQL with SSL/TLS and a Model Registry configured to use a secure DB connection.
+
+    Ensures the odh-trusted-ca-bundle ConfigMap is mounted in both MySQL and Model Registry deployments,
+    and the --ssl-ca argument is set to the correct path.
+    """
+    patch = {"spec": {"template": mysql_template_with_ca["spec"]}}
+
+    with ResourceEditor(patches={model_registry_db_deployment: patch}):
+        with create_model_registry_instance(
+            namespace=model_registry_namespace,
+            name=SECURE_MR_NAME,
+            labels=MODEL_REGISTRY_STANDARD_LABELS,
+            grpc={},
+            rest={},
+            istio=ISTIO_CONFIG_DICT,
+            mysql=model_registry_mysql_config,
+            wait_for_resource=True,
+        ) as mr:
+            mr.wait_for_condition(condition="Available", status="True")
+            yield mr
+
+
+@pytest.fixture()
+def local_ca_bundle(request: pytest.FixtureRequest, admin_client: DynamicClient) -> Generator[str, Any, Any]:
+    """
+    Creates a local CA bundle file by fetching the CA bundle from a ConfigMap and appending the router CA from a Secret.
+    """
+    namespace = getattr(request, "param", {}).get("namespace", py_config["model_registry_namespace"])
+    ca_bundle_path = getattr(request, "param", {}).get("ca_bundle_path", "ca-bundle.crt")
+    cert_name = getattr(request, "param", {}).get("cert_name", "ca-bundle.crt")
+    create_ca_bundle_file(
+        admin_client=admin_client, namespace=namespace, ca_bundle_path=ca_bundle_path, cert_name=cert_name
+    )
+    yield ca_bundle_path
+
+    try:
+        os.remove(ca_bundle_path)
```

Comment:
You can remove the try/except. Yiu are simply catching and raising the same exception